smack-experimental/src/main/java/org/jivesoftware/smackx/mam/MamManager.java

```diff
@@ -714,6 +721,26 @@ public boolean isSupported() throws NoResponseException, XMPPErrorException, Not
         return serviceDiscoveryManager.supportsFeature(archiveAddress, MamElements.NAMESPACE);
     }
 
+    public boolean isAdvancedConfigurationSupported(Jid jid) throws InterruptedException, XMPPException, SmackException {
+        DiscoverItems discoverItems = adHocCommandManager.discoverCommands(jid);
+        for (DiscoverItems.Item item : discoverItems.getItems()) {
+            if (item.getNode().equals(ADVANCED_CONFIG_NAMESPACE)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public List<RemoteCommand> getAdvancedConfigurationCommands(Jid jid) throws InterruptedException, XMPPException, SmackException {
+        List<RemoteCommand> remoteCommands = new ArrayList<>();
+        DiscoverItems discoverItems = adHocCommandManager.discoverCommands(jid);
+        for (DiscoverItems.Item item : discoverItems.getItems()) {
+            if (item.getNode().equals(ADVANCED_CONFIG_NAMESPACE))
+                remoteCommands.add(adHocCommandManager.getRemoteCommand(jid, item.getNode()));
```

Comment:
Can there be more than one item with the namespace? If not, then we do not need to return a list here.

Reply:
Changed to single object instead of list